### PR TITLE
Align ContentControl Behavior with UWP behavior for empty children.

### PR DESCRIPTION
**Pull request by GuillaumeSE**
GitHub Issue (If applicable): #

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

When a ContentControl in the Visual Tree does not have child elements, the dataContext is displayed instead of nothing like on UWP
## What is the new behavior?

Now when a ContentControl does not have content, the control will be empty.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](https://github.com/unoplatform/uno/blob/master/README.md#uno-features)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Updated the [Release Notes](https://github.com/unoplatform/uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs [hunk 1]
```diff
@@ -323,7 +333,7 @@ public void UpdateContentTemplateRoot()
 
 				if (Content != null
 					&& !(Content is View)
-					&& ContentTemplateRoot == null
+					&& ContentTemplateRoot == null && dataTemplate == null && ContentTemplate == null
```
Comment (MatFillion) [inline src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs]: I think you should split lines

**Comment by MatFillion** [comment]
We should build a few tests for this :)

**Comment by GuillaumeSE** [comment]
> We should build a few tests for this :)

Incoming, trying to make relevant tests :)

File: src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs [hunk 2]
```diff
@@ -67,9 +67,19 @@ public virtual object Content
...
+				}
+				else if(ContentTemplate != null)
+				{										 
+				   return DataContext;
+				}
+				else
+				{
+					return null;
+				}
+
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs]: Please add comment explaining each scenario.

Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs]: Also, I'm not sure if the original version was doing it properly as well. `GetValue(ContentControl.ContentProperty)` would return an incorrect value (e.g. in template binding scenarios).

We may need to change the local value instead, reacting do `DataContext`, `ContentTemplate` and `Content` changes.

Also, there are no tests for this scenario, could you add some? They can be unit tests, not UI tests.

**Pull request merged**